front/app/modules/commercial/insights/hooks/useInsightsView.ts

```diff
@@ -1,7 +1,7 @@
 import { useState, useEffect } from 'react';
 import { insightsViewStream, IInsightsView } from '../services/insightsViews';
 
-const useInsightsView = (id: string) => {
+const useInsightsView = (id: string | undefined) => {
```

Comment:
How can the id be undefined? This is not how we should handle this. Rather you should cast the type of the param when you access it:
```
const { viewId } = useParams() as {viewId: string};
```
Also mentioned in the new Router docs.

Reply:
yep, sorry, was super clear in the docs..  🤦🏻‍♂️🤦🏻‍♂️🤦🏻‍♂️